### PR TITLE
Generated a .NET SDK based on new api-version 2020-12-01

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by panchagnula** [comment]
//CC: @miwebst

**Review by panchagnula — APPROVED** [review]
Thanks @Kotasudhakarreddy

**Comment by JonathanCrd** [comment]
Hi, could you provide the link to the [azure rest api specs](https://github.com/Azure/azure-rest-api-specs) PR?

**Comment by panchagnula** [comment]
> Hi, could you provide the link to the [azure rest api specs](https://github.com/Azure/azure-rest-api-specs) PR?

See https://github.com/Azure/azure-rest-api-specs/pull/13144

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/SkuCapacity.cs
Comment (JonathanCrd) [inline sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/SkuCapacity.cs:42]: `int? elasticMaximum = default(int?)` should be after `scaleType` since it is a new parameter.

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/SlowRequestsBasedTrigger.cs
Comment (JonathanCrd) [inline sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/SlowRequestsBasedTrigger.cs:36]: `string path = default(string)` should be at the end since it is a new parameter.

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StackMajorVersion.cs
Comment (JonathanCrd) [inline sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StackMajorVersion.cs:66]: Same here with `appSettingsDictionary` and `siteConfigPropertiesDictionary`.

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StaticSiteARMResource.cs
Comment (JonathanCrd) [inline sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StaticSiteARMResource.cs:65]: Same here. `templateProperties`, `contentDistributionEndpoint`, `keyVaultReferenceIdentity`, and `userProvidedFunctionApps` should be after `SkuDescription sku = default(SkuDescription)`.

**Review by JonathanCrd — CHANGES_REQUESTED** [review]
Found some breaking changes in several files.
Most of them has to do with rearranging the parameters in the swagger file, so new parameters are at the end of the list.

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs
Comment (Kotasudhakarreddy) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: I think Yes. @panchagnula could you please confirm here.

Comment (panchagnula) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: @JonathanCrd  thanks for pointing this out. No this should not be removed. CC:// @pvbchak who did the rest-api-spec change this file is missing this API https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/web/resource-manager/Microsoft.Web/stable/2020-12-01/WebApps.json 

while this doesn't https://raw.githubusercontent.com/Azure/azure-rest-api-specs/master/specification/web/resource-manager/Microsoft.Web/stable/2020-10-01/WebApps.json

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StaticSiteARMResource.cs
Comment (panchagnula) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/Models/StaticSiteARMResource.cs]: @JonathanCrd while trying to make the changes to parameter order that was called out here - the swagger validation is flagging these as breaking changes https://github.com/Azure/azure-rest-api-specs/pull/14018/checks?check_run_id=2422311229 
how do we resolve this?

**Comment by JonathanCrd** [comment]
Changes in this PR were reverted, are you still pursuing to make changes and merge this PR?

Going to close it next week if not response.

**Comment by Kotasudhakarreddy** [comment]
> Changes in this PR were reverted, are you still pursuing to make changes and merge this PR?
> 
> Will be closed next week if not response.

@JonathanCrd yes we are on it. We will push new changes to this PR.

**Comment by Kotasudhakarreddy** [comment]
@JonathanCrd - I have followed the SDK generation process defined [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process). When I generate the SDK with the given steps I have seen lot of breaking changes in the files generated. I am looking for some help here to understand the generation process in a better way. 
Note: I have observed that the SDK file generation process has been changed from the last release(Sept 2020).

**Comment by panchagnula** [comment]
> @JonathanCrd - I have followed the SDK generation process defined [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process). When I generate the SDK with the given steps I have seen lot of breaking changes in the files generated. I am looking for some help here to understand the generation process in a better way.
> Note: I have observed that the SDK file generation process has been changed from the last release(Sept 2020).

@JonathanCrd is this behavior because .NET SDK is now using Track2? Is there a way we can get this generated using Track 1 so we can catch the //BUILD release?

**Comment by JonathanCrd** [comment]
Hi @Kotasudhakarreddy @panchagnula, the errors in the CI (Build) are caused because the generated files are being manually modified. The generated files should not me modified manually, instead, the breaking changes must be resolved directly from the swagger file. The wiki that Kota linked is up to date.

My understanding is that the generation script and the release process are not being affected by track 2 right now, but I'm going to double check on that.

You can also attend our team's office hours on Wednesdays to help you out with your questions.

**Comment by panchagnula** [comment]
> Hi @Kotasudhakarreddy @panchagnula, the errors in the CI (Build) are caused because the generated files are being manually modified. The generated files should not me modified manually, instead, the breaking changes must be resolved directly from the swagger file. The wiki that Kota linked is up to date.
> 
> My understanding is that the generation script and the release process are not being affected by track 2 right now, but I'm going to double check on that.
> 
> You can also attend our team's office hours on Wednesdays to help you out with your questions.

@JonathanCrd thank you for the response.  @Kotasudhakarreddy I believe you re-generated the swagger using the new updates to rest-api-spec  https://github.com/Azure/azure-rest-api-specs/pull/14018 - & were seeing some changes in new files being generated? Did you push the new changes to this branch after you re-tried. If not do that so that @JonathanCrd can review & help unblock here. Thanks!

**Comment by JonathanCrd** [comment]
The metadata text file is missing in the PR. (This file can be found [here](https://github.com/Azure/azure-sdk-for-net/pull/20300/commits/ef58c916f176a382a5a12044350ee6bc73be8ab0#diff-dd8a7b891d62bf1584ade9d3637b86893ea1bf05cda9fdad64d5917a6f2be236), in the first commit in this PR)

Please rerun the generate.ps1 script, this will add the metadata file. 
After that we can take a look at the changes.

**Comment by Kotasudhakarreddy** [comment]
> The metadata text file is missing in the PR. (This file can be found [here](https://github.com/Azure/azure-sdk-for-net/pull/20300/commits/ef58c916f176a382a5a12044350ee6bc73be8ab0#diff-dd8a7b891d62bf1584ade9d3637b86893ea1bf05cda9fdad64d5917a6f2be236), in the first commit in this PR)
> 
> Please rerun the generate.ps1 script, this will add the metadata file.
> After that we can take a look at the changes.

I am regenerating it. Let me push the latest changes

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/CertificatesOperations.cs
Comment (JonathanCrd) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/CertificatesOperations.cs:84]: This is still a breaking change

**Comment by JonathanCrd** [comment]
> @JonathanCrd Jonathan Cardenas FTE could you please review.
> 
> @panchagnula Sisira Panchagnula FTE I have generated the SDK with latest changes from Swagger.

Hi, I re-review this PR and found that some breaking changes are still present. In order to solve this, the order of the parameters should be changed directly in the swagger and not manually editing the generated files.

**Comment by Kotasudhakarreddy** [comment]
@JonathanCrd - Thanks for your review. Suggested changes have been made and merged in to the swagger . [here](https://github.com/Azure/azure-rest-api-specs/pull/14018) is the PR link for the same. But when I use [generate.ps1](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/websites/Microsoft.Azure.Management.WebSites/src/generate.ps1) to generate the SDK with latest changes merged to [swagger/rest-api-spec](https://github.com/azure/azure-rest-api-specs) , the changes are not reflected . Is there a way to generate the SDK pointing to specific Commit/Merge.

**Comment by Kotasudhakarreddy** [comment]
@JonathanCrd - can I get some help here.

File: sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs
Comment (JonathanCrd) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: Hi @panchagnula , @pvbchak, @Kotasudhakarreddy:
This is still unresolved, should this method not be removed?

We are considering this to be the last remaining issue for this PR, once resolved we can merge it.

Comment (Kotasudhakarreddy) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: @panchagnula - I think `CreateOrUpdateSwiftVirtualNetworkConnection` has been renamed to  `CreateOrUpdateSwiftVirtualNetworkConnectionWithCheck`. please confirm.

Comment (Kotasudhakarreddy) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: @JonathanCrd - This method has been renamed to `CreateOrUpdateSwiftVirtualNetworkConnectionWithCheck`. please see the PR link [here](https://github.com/Azure/azure-rest-api-specs/pull/14018/commits/7cf7add97ddda5d87b1b6fbe2e575d86688ae11e). So this is expected.

Comment (JonathanCrd) [reply sdk/websites/Microsoft.Azure.Management.WebSites/src/Generated/WebAppsOperationsExtensions.cs:6133]: Got it, thank you for the clarification.

**Review by JonathanCrd — APPROVED** [review]

**Pull request merged**
